crm-project/src/main/java/CRM/service/UserService.java

```diff
@@ -140,50 +143,64 @@ public List<User> getAllInBoard(long boardId) throws AccountNotFoundException {
-     * @param boardId the id of the board to add the user to
-     * @return the UserInBoard object representing the user being added to the board
-     * @throws AccountNotFoundException if the user or board with the given id does not exist in the database
-     * @throws IllegalArgumentException if the combination of the given user and board already exists in the database
-     */
-    public void addUserToBoard(long userId, long boardId) throws AccountNotFoundException {
+    //TODO documentation
+    public List<User> updateUserToBoard(ObjectsIdsRequest objectsIdsRequest) throws AccountNotFoundException {
         User user;
         Board board;
         try {
-            user = Validations.doesIdExists(userId, userRepository);
-            board = Validations.doesIdExists(boardId, boardRepository);
-
+            user = Validations.doesIdExists(objectsIdsRequest.getUserId(), userRepository);
+            board = Validations.doesIdExists(objectsIdsRequest.getBoardId(), boardRepository);
         } catch (NoSuchElementException e) {
             throw new AccountNotFoundException(ExceptionMessage.ACCOUNT_DOES_NOT_EXISTS.toString());
         }
 
-        NotificationSetting notificationSetting = Validations.doesIdExists(2L, settingRepository);
-
-        UserPermission userPermission = new UserPermission();
-        userPermission.setId(0L);
-        userPermission.setUser(user);
-        userPermission.setPermission(Permission.USER);
+        if(user.equals(board.getCreatorUser())){
+            throw new IllegalArgumentException(ExceptionMessage.ADMIN_CANT_CHANGE_HIS_PERMISSION.toString());
+        }
 
-        board.addUserPermissionToBoard(userPermission);
+        Permission permissionRequest = Permission.values()[Math.toIntExact(objectsIdsRequest.getPermissionId())];
+        Set<UserPermission> userPermissionsSet = updateUserPermission(user, permissionRequest, board);
+        List<User> users = board.getAllUsersInBoard(board, userPermissionsSet);
 
-        createDefaultSettingForNewUserInBoard(user, board, notificationSetting);
         boardRepository.save(board);
+        return users;
     }
 
-    /**
-     * Creates default notifications for every new user in every board,
-     * using constant notifications
-     */
-    private void createDefaultSettingForNewUserInBoard(User user, Board board, NotificationSetting notificationSetting) {
-        UserSetting userSetting = new UserSetting();
-        userSetting.setId(0L);
-        userSetting.setInApp(true);
-        userSetting.setInEmail(true);
-        userSetting.setUser(user);
-        userSetting.setSetting(notificationSetting);
-        board.addUserSettingToBoard(userSetting);
+    //TODO documentation
+    private void createDefaultSettingForNewUserInBoard(User user, Board board) {
+        for (long i = 0; i < 6; i++) {
+            NotificationSetting notificationSetting = Validations.doesIdExists(i+1L, settingRepository);
```

Comment:
That's not right. We have "settingsRepository". Use it to get all the settings in the database. What happens if we decide to remove 2 settings? We wouldn't be able to create users.
So get all the notifications from the database, and based on their's list's length, loop it and add them to the user

This way you don't have to use "doesIdExist" function

Reply:
I dont handle setting feature in first place and i didnt wrote the line before that inserted notification with id 2L